x-pack/plugin/mapper-patterned-text/src/main/java/org/elasticsearch/xpack/patternedtext/PatternedTextFieldType.java

```diff
@@ -0,0 +1,267 @@
+        );
+        IntervalsSource fuzzyIntervals = Intervals.multiterm(fuzzyQuery.getAutomata(), IndexSearcher.getMaxClauseCount(), term);
+        return toIntervalsSource(fuzzyIntervals, fuzzyQuery, context);
+    }
+
+    @Override
+    public IntervalsSource wildcardIntervals(BytesRef pattern, SearchExecutionContext context) {
+        return toIntervalsSource(
+            Intervals.wildcard(pattern, IndexSearcher.getMaxClauseCount()),
+            new MatchAllDocsQuery(), // wildcard queries can be expensive, what should the approximation be?
+            context
+        );
+    }
+
+    @Override
+    public IntervalsSource regexpIntervals(BytesRef pattern, SearchExecutionContext context) {
+        return toIntervalsSource(
+            Intervals.regexp(pattern, IndexSearcher.getMaxClauseCount()),
+            new MatchAllDocsQuery(), // regexp queries can be expensive, what should the approximation be?
+            context
+        );
+    }
+
+    @Override
+    public IntervalsSource rangeIntervals(
+        BytesRef lowerTerm,
+        BytesRef upperTerm,
+        boolean includeLower,
+        boolean includeUpper,
+        SearchExecutionContext context
+    ) {
+        return toIntervalsSource(
+            Intervals.range(lowerTerm, upperTerm, includeLower, includeUpper, IndexSearcher.getMaxClauseCount()),
+            new MatchAllDocsQuery(), // range queries can be expensive, what should the approximation be?
+            context
+        );
+    }
+
+    @Override
+    public Query phraseQuery(TokenStream stream, int slop, boolean enablePosIncrements, SearchExecutionContext queryShardContext)
+        throws IOException {
+        final Query textQuery = textFieldType.phraseQuery(stream, slop, enablePosIncrements, queryShardContext);
+        return sourceConfirmedQuery(textQuery, queryShardContext);
+    }
+
+    @Override
+    public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements, SearchExecutionContext queryShardContext)
+        throws IOException {
+        final Query textQuery = textFieldType.multiPhraseQuery(stream, slop, enablePositionIncrements, queryShardContext);
+        return sourceConfirmedQuery(textQuery, queryShardContext);
+    }
+
+    @Override
+    public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions, SearchExecutionContext queryShardContext)
+        throws IOException {
+        final Query textQuery = textFieldType.phrasePrefixQuery(stream, slop, maxExpansions, queryShardContext);
+        return sourceConfirmedQuery(textQuery, queryShardContext);
+    }
+
+    @Override
+    public BlockLoader blockLoader(BlockLoaderContext blContext) {
+        return new BlockDocValuesReader.BytesRefsFromBinaryBlockLoader(name());
+    }
+
+    @Override
+    public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+        var templateDataBuilder = new SortedSetOrdinalsIndexFieldData.Builder(
```

Comment:
I think for now we can wait adding support for field data. This is similar to `match_only_text`. So maybe remove this method and `PatternedTextIndexFieldData `?

I think we just need field data support for the template leaf field. But we can add that in a follow up.